packages/flutter/lib/src/foundation/memory_allocations.dart

```diff
@@ -82,6 +126,9 @@ class ObjectCreated extends ObjectEvent {
   }
 }
 
+/// Callback for creating an object of type [T].
+typedef ObjectBuilderCallback<T> = T Function();
```

Comment:
https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#leave-breadcrumbs-in-the-comments

> Typedefs should mention at least one place where the signature is used.